src/Nethermind/Nethermind.Network/P2P/Analyzers/MetricsDisconnectsAnalyzer.cs

```diff
@@ -11,42 +12,44 @@ public void ReportDisconnect(DisconnectReason reason, DisconnectType type, strin
         {
             if (type == DisconnectType.Remote)
             {
-                switch (reason)
+                Metrics.RemoteDisconnectsTotal.Increment(reason);
+
+                switch (reason.ToEthDisconnectReason())
```

Comment:
minor: You could move the conversion `reason.ToEthDisconnectReason()` before both if's

Reply:
Done